src/util/spreadsheet/misc.py

```diff
@@ -1,32 +1,46 @@
-from .. import json
+from .. import json, pg
+
+MEMORY_CAP = 2 * 10**8  # 200Mo
 
 
 def iter_commands(cr, like_all=(), like_any=()):
     if not (bool(like_all) ^ bool(like_any)):
         raise ValueError("Please specify `like_all` or `like_any`, not both")
-    cr.execute(
-        """
-        SELECT id,
-               commands
-          FROM spreadsheet_revision
-         WHERE commands LIKE {}(%s::text[])
-        """.format("ALL" if like_all else "ANY"),
-        [list(like_all or like_any)],
-    )
-    for revision_id, data in cr.fetchall():
-        data_loaded = json.loads(data)
-        if "commands" not in data_loaded:
-            continue
-        data_old = json.dumps(data_loaded, sort_keys=True)
-
-        changed = yield data_loaded["commands"]
-        if changed is None:
-            changed = data_old != json.dumps(data_loaded, sort_keys=True)
-
-        if changed:
-            cr.execute(
-                "UPDATE spreadsheet_revision SET commands=%s WHERE id=%s", [json.dumps(data_loaded), revision_id]
+
+    with pg.named_cursor(cr) as ncr:
+        ncr.execute(
+            """
+            WITH buckets AS (
+                SELECT id,
+                       SUM(LENGTH(commands)) OVER (ORDER BY id) / {memory_cap} AS num,
+                       commands
+                  FROM spreadsheet_revision
+                 WHERE commands LIKE {condition}(%s::text[])
+              ORDER BY id
             )
+            SELECT ARRAY_AGG(id ORDER BY id),
+                   ARRAY_AGG(commands ORDER BY id)
+              FROM buckets
+          GROUP BY num
```

Comment:
This is fine but it has the downside that if you have a very large record you will get it along with all other smaller records that fit in the last bucket.
Example
```
|--butket1--|--butket1--|--butket2--|
***big*record*************
                          *small1
                           *small2
                            ...
                                  *smallN
```
Total size in group associated to bucket 3: 3*bucket_size, whether we could have gotten _just_ big_record's size if we left out the smaller ones. It would depend on whether we actually get cases where there is a HUGE record that is still OK alone but too much with the extra ~200MB.

All this to say that's why I added the `len > memory_cap AS alone` then use it in the groping ;)